app/javascript/packs/pages/edit_provider/EditProviderForm.jsx

```diff
@@ -0,0 +1,243 @@
+import React from "react";
+import Typography from "@material-ui/core/Typography";
+import Grid from "@material-ui/core/Grid";
+import { makeStyles } from "@material-ui/core/styles";
+import TextField from "@material-ui/core/TextField";
+import InputLabel from "@material-ui/core/InputLabel";
+import Select from "@material-ui/core/Select";
+import FormControl from "@material-ui/core/FormControl";
+import FormHelperText from "@material-ui/core/FormHelperText";
+import Checkbox from "@material-ui/core/Checkbox";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import FormLabel from "@material-ui/core/FormLabel";
+import STATES from "../../data/states";
+import providerRequestTypes from "../../data/providerRequestTypes";
+import Button from "@material-ui/core/Button";
+import Switch from "@material-ui/core/Switch";
+import  {
+  TOKEN_ENTITY_REQUEST_STATES
+} from "../../hooks/useTokenEntity";
+
+const useStyles = makeStyles(theme => ({
+  mediumSpacing: {
+    marginBottom: "30px"
+  },
+  largeSpacing: {
+    marginBottom: "50px"
+  }
+}));
+
+const EditProviderForm = ({provider, onChange, setField, requestState, saveProvider}) => {
+  const classes = useStyles();
+  return (
+    <>
+      <Grid item xs={12} className={classes.mediumSpacing}>
+        <FormControl>
+          <FormLabel required id="type-select-label">
+            Request visibility
+          </FormLabel>
+          <FormControlLabel
+            control={
+              <Switch
+                checked={provider.active}
+                onChange={e => {
+                  setField("active")(e.target.checked);
+                }}
+                name="active"
+                color="primary"
+              />
+            }
+            label={provider.active ? "Published" : "Unpublished"}
+          />
+        </FormControl>
+      </Grid>
+      <Grid item xs={12} className={classes.mediumSpacing}>
+        <FormControl>
+          <FormLabel required id="type-select-label">
+            What kinds of support do you need?
+          </FormLabel>
+          {providerRequestTypes.map(type => (
+            <FormControlLabel
+              key={type.value}
+              control={
+                <Checkbox
+                  checked={provider.requests.includes(type.value)}
+                  onChange={e =>
+                    e.target.checked
+                      ? setField("requests")([
+                          ...provider.requests,
+                          type.value
+                        ])
+                      : setField("requests")(
+                          provider.requests.filter(x => x !== type.value)
+                        )
+                  }
+                  name={type.value}
+                  color="primary"
+                />
+              }
+              label={type.label}
+            />
+          ))}
+        </FormControl>
+      </Grid>
+      <Grid item xs={12} className={classes.mediumSpacing}>
```

Comment:
Probably not worth worrying about for now, but eventually we may want to unify this form with the signup form.